Bio/SeqUtils/__init__.py

```diff
@@ -11,31 +11,136 @@
+    """Calculate G+C percentage in seq (float between 0 and 1).
+
+    Copes with mixed case sequences. Ambiguous Nucleotides in this context are
+    those different from ATCGSW.
+
+    If ambiguous equals "ignore" (default) , it will treat only unambiguous
+    nucleotides (GCS) as counting towards the GC percentage.
+
+    If ambiguous equals "count", will use a "mean" value when counting the
+    ambiguous characters, for example, G and C will still be counted as 1, N
+    and X will be counted as 0.5, D will be counted as 0.33 etc. See
+    Bio.SeqUtils._ambiguous_gc_values for a full list.
+
+    If ambiguous equals "remove", will only count GCS towards the percentage,
+    but will also only count ACTGSW towards the length of the sequence.
+    Equivalent to gc_content(seq.replace('N','')) but replacing all nucleotides
+    that are ambiguous between GC or AT (VBMRYKXNHD).
+
+    Will raise a ValueError for any other value of the ambiguous parameter.
+
+
+    >>> from Bio.SeqUtils import gc_content
+    >>> seq = "ACTG"
+    >>> print(f"GC content of {seq} : {gc_content(seq):.2f}")
+    GC content of ACTG : 0.50
+
+    S and W are not considered ambiguous.
+
+    >>> seq = "ACTGSSSS"
+    >>> print(f"GC content of {seq} : {gc_content(seq):.2f}")
+    GC content of ACTGSSSS : 0.75
+    >>> gc = gc_content(seq, "count")
+    >>> print(f"GC content with ambiguous counting: {gc:.2f}")
+    GC content with ambiguous counting: 0.75
+
+    Some examples with ambiguous nucleotides.
+
+    >>> seq = "ACTGN"
+    >>> print(f"GC content of {seq} : {gc_content(seq):.2f}")
+    GC content of ACTGN : 0.40
+    >>> gc = gc_content(seq, "count")
+    >>> print(f"GC content with ambiguous counting: {gc:.2f}")
+    GC content with ambiguous counting: 0.50
+    >>> gc = gc_content(seq, "remove")
+    >>> print(f"GC content with ambiguous removing: {gc:.2f}")
+    GC content with ambiguous removing: 0.50
+
+    Ambiguous nucleotides are also removed from the length of the sequence.
+
+    >>> seq = "GDVV"
+    >>> print(f"GC content of {seq} : {gc_content(seq):.2f}")
+    GC content of GDVV : 0.25
+    >>> gc = gc_content(seq, "count")
+    >>> print(f"GC content with ambiguous counting: {gc:.4f}")
+    GC content with ambiguous counting: 0.6667
+    >>> gc = gc_content(seq, "remove")
+    >>> print(f"GC content with ambiguous removing: {gc:.2f}")
+    GC content with ambiguous removing: 1.00
 
-def GC(seq):
-    """Calculate G+C content, return percentage (as float between 0 and 100).
 
-    Copes mixed case sequences, and with the ambiguous nucleotide S (G or C)
-    when counting the G and C content.  The percentage is calculated against
-    the full length, e.g.:
+    Note that this will return zero for an empty sequence.
+    """
+    if ambiguous not in ("count", "remove", "ignore"):
+        raise ValueError(f"ambiguous value '{ambiguous}' not recognized")
 
-    >>> from Bio.SeqUtils import GC
-    >>> GC("ACTGN")
-    40.0
+    count = Counter(seq)
```

Comment:
Surprisingly, I am finding that `seq.count(x) for x in "ACGTacgt"` is about ten times faster than `Counter(seq)`.

Reply:
Yeah, I think I choose it only looking from a code clarity point of view (that I don't even know holds up looking at it now), but the performance loss is too big, sorry for not catching it. Will update